pkg/controller/llmisvc/controller.go

```diff
@@ -109,25 +111,33 @@ func (r *LLMInferenceServiceReconciler) Reconcile(ctx context.Context, req ctrl.
 		r.Recorder.Eventf(original, corev1.EventTypeWarning, "Error", "Reconciliation failed: %v", reconcileErr.Error())
 	}
 
-	if err := r.updateStatus(ctx, original, resource); err != nil {
+	if err := r.updateStatus(ctx, resource); err != nil {
 		logger.Error(err, "Failed to update status for LLMInferenceService")
 		return ctrl.Result{}, err
 	}
 
 	return ctrl.Result{}, reconcileErr
 }
 
-func (r *LLMInferenceServiceReconciler) updateStatus(ctx context.Context, existing *v1alpha1.LLMInferenceService, desired *v1alpha1.LLMInferenceService) error {
-	if equality.Semantic.DeepEqual(existing.Status, desired.Status) {
-		// If there's nothing to update, just return.
-		return nil
-	}
+func (r *LLMInferenceServiceReconciler) updateStatus(ctx context.Context, desired *v1alpha1.LLMInferenceService) error {
+	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		latest := &v1alpha1.LLMInferenceService{}
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(desired), latest); err != nil {
+			return err
+		}
```

Comment:
Wanted to raise it a separate PR, but it landed here. I noticed that sometimes reconcile fails on conflict when trying to update status - wrapped it in retry but I wonder if there's more appropriate fix for that @pierDipi.

Reply:
it's less "hope cache is up to date now"

Reply:
> it's less "hope cache is up to date now"

I know :)

Reply:
> I can try to eliminate first GET

perhaps as a follow-up? it's optimization after all